### PR TITLE
Allow background color and n_colors for the scalarbar

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR allows the `add_scalar_bar()` function to use a given `background_color` in the very specific case where the input colormap uses alpha channel. It also adds a new `n_colors` parameter to control the number of colors in the bar.

```py
import numpy as np
import pyvista as pv
from matplotlib.colors import ListedColormap
arr = np.linspace([1.0, 0.7, 0.5, 1.],
                  [0., 0., 0., 0.],
                  128)
colors = np.concatenate((arr, np.flip(arr)))
cmap = ListedColormap(colors)

p = pv.Plotter()
p.background_color = 'black'
p.add_mesh(pv.Cone().elevation(), cmap=cmap)
p.add_scalar_bar(title='Elevation (with background_color)',
                 background_color=(0.5, 0.5, 0.5), n_colors=256)
p.show()
```

image

### CODE REVIEW

**Comment by akaszynski** [comment]
Awesome!  Thanks!

File: pyvista/plotting/plotting.py [hunk 1]
```diff
@@ -2166,7 +2173,31 @@ def add_scalar_bar(self, title=None, n_labels=5, italic=False,
 
         # Create scalar bar
         self.scalar_bar = vtk.vtkScalarBarActor()
-        self.scalar_bar.SetLookupTable(mapper.lookup_table)
+        if background_color is not None:
+            from ..core.common import vtk_to_numpy, numpy_to_vtk
+            if not isinstance(background_color, collections.Iterable):
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py]: We should leverage `parse_color` here incase the user passes a string name of a color to be consistent with the rest of the API

File: pyvista/plotting/plotting.py [hunk 2]
```diff
@@ -2166,7 +2173,31 @@ def add_scalar_bar(self, title=None, n_labels=5, italic=False,
 
         # Create scalar bar
         self.scalar_bar = vtk.vtkScalarBarActor()
-        self.scalar_bar.SetLookupTable(mapper.lookup_table)
+        if background_color is not None:
+            from ..core.common import vtk_to_numpy, numpy_to_vtk
```
Comment (banesullivan) [inline pyvista/plotting/plotting.py]: We should have an absolute import for this

**Comment by banesullivan** [comment]
Thanks for adding these new features @GuillaumeFavelier! There are a few concerns I have which I will clean up on the master branch

**Pull request merged**